### PR TITLE
Feat: add chat-gpt workflow step and upgrade workflow version

### PR DESCRIPTION
**Pull request by FogDong**
### Checklist

I have:

- [ ] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [ ] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [ ] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

####  Verified Addon promotion rules

If this pr wants to promote an experimental addon to verified, you must check whether meet these conditions too:
  - [ ] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.
  - This addon must have some basic but necessary information.
    - [ ] An accessible icon url and source url defined in addon's `metadata.yaml`.
    - [ ] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
    - [ ] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    - [ ] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

File: addons/vela-workflow/definitions/chat-gpt.cue [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		"definition.oam.dev/example-url": "https://raw.githubusercontent.com/kubevela/workflow/main/examples/workflow-run/chat-gpt.yaml"
+		"category":                       "External Intergration"
+	}
+	labels: {}
+	type: "workflow-step"
+}
+
+template: {
+	token: op.#Steps & {
+		if parameter.token.value != _|_ {
```
Comment (barnettZQG) [inline addons/vela-workflow/definitions/chat-gpt.cue:21]: Can we add a config template to configure the token?

File: addons/vela-workflow/definitions/chat-gpt.cue [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+	response: json.Unmarshal(http.response.body)
+	fail:     op.#Steps & {
+		if http.response.statusCode >= 400 {
+			requestFail: op.#Fail & {
+				message: "\(http.response.statusCode): failed to request: \(response.error.message)"
+			}
+		}
+	}
+	result: response.choices[0].message.content
+	log:    op.#Log & {
```
Comment (barnettZQG) [inline addons/vela-workflow/definitions/chat-gpt.cue:104]: Could you only write the log? What things can we do using this step?

Comment (FogDong) [reply addons/vela-workflow/definitions/chat-gpt.cue:104]: https://github.com/kubevela/workflow/blob/main/examples/workflow-run/chat-gpt.yaml

**Comment by wangyikewxgm** [comment]
Could you add an example for this capability?

**Comment by FogDong** [comment]
@wangyikewxgm The example is shown in the definition: 
```
"definition.oam.dev/example-url": "https://raw.githubusercontent.com/kubevela/workflow/main/examples/workflow-run/chat-gpt.yaml"
```

**Review by wangyikewxgm — APPROVED** [review]

**Review by barnettZQG — APPROVED** [review]

**Pull request merged**